### PR TITLE
fix: spop set index get error

### PR DESCRIPTION
**Pull request by Zzhiter**
fix issue: https://github.com/microsoft/garnet/issues/190

### CODE REVIEW

File: test/Garnet.test/RespSetTest.cs [hunk 1]
```diff
@@ -388,6 +388,19 @@ public void CanDoSPOPWithCountCommandLC()
             Assert.AreEqual(expectedResponse, strResponse);
         }
 
+        [Test]
+        public void CanRemoveAllElementsLC()
+        {
+            var myset = new HashSet<string> { "one", "two", "three", "four", "five" };
```
Comment (TalZaccai) [inline test/Garnet.test/RespSetTest.cs:394]: Doesn't look like this variable is needed...

File: test/Garnet.test/RespSetTest.cs [hunk 2]
```diff
@@ -388,6 +388,19 @@ public void CanDoSPOPWithCountCommandLC()
             Assert.AreEqual(expectedResponse, strResponse);
         }
 
+        [Test]
+        public void CanRemoveAllElementsLC()
```
Comment (TalZaccai) [inline test/Garnet.test/RespSetTest.cs:392]: I would add this case to the existing CanDoSPOPWithCountCommandLC test

**Review by TalZaccai — CHANGES_REQUESTED** [review]

File: test/Garnet.test/RespSetTest.cs (hunk 2 above)
Comment (Zzhiter) [reply test/Garnet.test/RespSetTest.cs]: Thank you for your review, I think I understand what you mean and have fixed it

**Review by TalZaccai — CHANGES_REQUESTED** [review]

File: test/Garnet.test/RespSetTest.cs [hunk 3]
```diff
@@ -368,26 +368,31 @@ public void CanDoSPOPCommandLC()
...
-            myset.Add("four");
-            myset.Add("five");
-
             CreateLongSet();
 
-            using var lightClientRequest = TestUtils.CreateRequest();
-            var response = lightClientRequest.SendCommand("SPOP myset 3", 4);
+            var lightClientRequest = TestUtils.CreateRequest();
+            var response = lightClientRequest.SendCommand("SPOP myset 3", 3);
             var strLen = Encoding.ASCII.GetString(response).Substring(1, 1);
```
[the remark this answers is not in the document]
Comment (Zzhiter) [reply test/Garnet.test/RespSetTest.cs]: done

**Comment by Zzhiter** [comment]
It seems that all the problems have been solved, thank you for your review

**Review by TalZaccai — APPROVED** [review]

**Pull request merged**